include/ignition/rendering/BoundingBoxCamera.hh

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+#ifndef IGNITION_RENDERING_BOUNDINGBOXCAMERA_HH_
+#define IGNITION_RENDERING_BOUNDINGBOXCAMERA_HH_
+
+#include <cstdint>
+#include <vector>
+
+#include <ignition/common/Event.hh>
+#include <ignition/math/Color.hh>
+#include <ignition/math/Vector3.hh>
+
+#include "ignition/rendering/Camera.hh"
+
+namespace ignition
+{
+  namespace rendering
+  {
+    /// \brief BoundingBox types for Visible / Full 2D Boxes / 3D Boxes
+    enum class BoundingBoxType
+    {
+      /// 2D box that shows the full box of occluded objects
+      BBT_FULLBOX2D = 0,
+
+      /// 2D box that shows the visible part of the
+      /// occluded object
+      BBT_VISIBLEBOX2D = 1,
+
+      /// 3D oriented box
+      BBT_BOX3D = 2
+    };
+
+    /// \brief 2D or 3D Bounding box. It stores the
+    /// position / orientation / size info of the box and its label
+    struct BoundingBox
+    {
+      /// \brief Box type
+      ignition::rendering::BoundingBoxType type;
```

Comment:
Does the bounding box really need a type? Shouldn't the type belong to the camera?

Reply:
The `BoundingBox` class is now on a separate file, so if the types get defined on this header, using it there would create a circular dependency

Reply:
> Does the bounding box really need a type? Shouldn't the type belong to the camera?

I think you're right. The type should belong to the camera. I say we remove the `type` from the `BoundingBox` class for now. The nice thing is that since you modified this class to use PIMPL, if we find for whatever reason that we need to add this field back in later, we should be able to do so without issues.